### DISCUSSION TITLE
"can't access property" in x-for when underlying data are modified.

### DISCUSSION THREAD

**Discussion by giovvv** [alpinejs/alpine#3917] [1. Help] [answered]
I am not exactly sure if it is an Alpine.js bug or if I have deeply misunderstood something, but I get a mysterious "Alpine Expression Error: can't access property" warning (followed by an error) in the following example. (Since I am not sure where the problem exactly lies, the title of this post is itself a bit of a guess).

For context, the code is part of from a complex search form, with two tabs and where the user can add or remove search filters which contain select menus whose content depends on the filter type. I have removed all the code unrelated to the problem, and you can see the distilled example in the following pen:

https://codepen.io/giovvv-the-flexboxer/pen/jOdRqGN

The issue is reproducible in the following way. If the user checks the radio button, in any sequence, nothing bad happens. However, it fails in the following scenario: (1) click the "add filter" button (a select menu appears); (2) click a number of times the radio buttons (the select menu is removed). After an apparently random number of clicks on the radio buttons (usually 3-4), the thing explodes.

Since the absence of a proper ":key" attribute is said to cause strange behaviors, I have tried to include a key which is different for each item (generated at line 28, used at line 47) but it does not seem to solve the problem.

Any idea?
Thanks

**Comment by ekwoka** [comment, marked as answer]
I'm working on a fix for this issue, But broadly, like in this example, it's because you're depending on values that you change that don't apply to all of the children.

here you produce a context where the effects have tab and a desynced, which should generally be avoided to begin with.

For example, if you instead do 

```html
      <template x-for="[a,b] in Object.entries(available_filters[tab])" :key="a">
        <option :value="a" x-text="b.name"></option>
      </template>
```

No such issue could possibly occur.

**Reply by giovvv** [reply]
Thanks!
